### PR TITLE
Data storage

### PR DESCRIPTION
**Pull request by amon-cofie**
### Changes:

- Added the SaveData module in which we created the methods:

> save_books
> save_people
> save_rentals
> save

- Added the LoadData module in which we created the methods:

> load_books
> load_people
> load_rentals
> get_person
> get_book

### CODE REVIEW

File: io_p.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    rentals_data = JSON.generate(@rentals, { max_nesting: false })
+    File.write('./data_file/rentals.json', rentals_data)
+  end
+
+  def save()
+    save_books
+    save_people
+    save_rentals
+  end
+end
```
Comment (Whoistolu) [inline io_p.rb]: - [ ] [REQUIRED] Great job team 👍🏽 Unfortunately, it looks like the save module has some issues you need to look into. Whenever I exit the app, I get the error below:
Screenshot 2023-01-12 at 14 17 29
Kindly fix this and ask for another review.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,
@amon-cofie and @eerapheal 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: io_p.rb (hunk 1 above)
Comment (amon-cofie) [reply io_p.rb]: Thanks for the feedback, we will make fix the issue as soon as possible

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Team, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggestions.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "Bug fix".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**